### DISCUSSION TITLE
Accommodating our 10-year commitment to reproducibility of GxP insights as we transition from proprietary to open source backboned data science

### DISCUSSION THREAD

**Discussion by epijim** [rinpharma/positconf-roundtables-2023#1] [Ideas]
Many companies are working to embrace open source, and a shift to fully open sourced studies - but a clinical trial can run for 5+ years, and a health authority may request further work after a trial closes. 

This has led to three types of work being present:

- completely next gen work (e.g. studies 100% R/Python)
- mixed studies (e.g. initial work in SAS, later work in R/Python, or a split where data work in SAS, later reporting event in R)
- legacy studies (e.g. 100% SAS)

Ensuring all three are possible has important impacts on people and platforms, and brings complexity to our roadmaps. What can we learn from each other enabling these workflows?

**Comment by MichaelRimler** [comment]
For this one - who is the 'our'? Is this a Roche perspective, an r-in-pharma perspective, or wider?

**Reply by epijim** [reply]
The title is verbatim copied from the proposal, and the copy underneath was me trying to kick start the discussion - and I don't think we discussed what 'our' was scoped to in the title (tagging @harveyl888 and @philbowsher). But I'd take a stab saying they are phrased from the perspective of people from pharma attending the round tables. 

Another angle is maybe not focussing on languages, but the way of working changes and this has impacts on what you can do in a single platform. As 2 examples: 1) versioning and dev vs prod may have been by a propriety interface in legacy, and git in next gen. 2) Legacy macros may have been written to work within propriety tooling, while next-gen tools maybe more general and portable (e.g. from the Roche perspective`admiral` and `tern` can be used anywhere and easily deployed from Posit package manger into compute by users, while our SAS macros pre-admiral are not portable).

From discussions in places like the Programming Heads' SCE council, I think there is diversity where some companies are tackling this, while others are avoiding it by not transitioning in the first place. e.g. they have a clinical reporting platform, then they have a separate data science platform. 

@MichaelRimler - do you have any ideas to improve it? Either the title, or description?

**Reply by MichaelRimler** [reply]
I could see something like:
**What does a 10-year commitment to reproducibility of GxP insights with open source backboned data science, transitioning from the typical proprietary backbone, look like?**
It feels more inclusive of those not quite there yet - and give bandwidth for those that are more committed to this vision to be quite vocal and opinionated.

FWIW - I think the description is fine as-is.

**Comment by MikeKSmith-Pfizer** [comment]
From a business perspective, this transition from SAS towards open-source is not the cost-saving that some might imagine. "We can turn off SAS" isn't going to happen any time soon, because of legacy work. But the long term ROI is better, because we reduce our reliance on a single software provider for submission work. 

I think acknowledging this, makes it a good topic for discussion.

**Reply by kkmann** [reply]
Focusing on the governance and technical challenges of very long term reproducibility in an system involving open source could be very helpful. The complexities of very long-term reproducibility are easily overlooked.

**Comment by minhonglee** [comment]
I would look at this topic with a software agnostic approach.  As FDA and other regulatory agencies state that any software is acceptable use as long as there are sufficient controls in place to ensure that the results are reliable, we as sponsors can also look at how we use any software (R, Python, SAS, etc) and implement processes, controls, and tools that supports the generation of reliable results.     

In a practical sense, this would mean defining those high level principles that aligns with regulatory guidance and building processes and tools that are integrated together to ensure those principles are followed.  For example, user environments are configured to restrict the ability to use R packages that have not been tested.  Processes can be defined to require the use of utilities that log outputs and ensure only appropriate packages are used.  Reproducibility over an extended period of time takes all components (infrastructure, process, tools) working together. 

All of these tools and processes probably exist today or can be developed but should be done in a way that's either software agnostic or in a way that's extensible so that new software capabilities can be incorporated if needed.  It would be matter of thoughtfully integrating everything together so that they can all work together consistently over time and meet that commitment of reproducibility over 10 year period.
